### ISSUE TITLE
pydocstyle is no longer maintened and has isssues with overload

### ISSUE THREAD

**Issue by fzimmermann89** [PTB-MR/mrpro#185]
We are currently using pydocstyle in our precommit.
This is deprecated https://github.com/PyCQA/pydocstyle/pull/658 and, more importantly, has a number of issues with 
`@overload`. https://github.com/PyCQA/pydocstyle/issues/613, https://github.com/PyCQA/pydocstyle/issues/525, https://github.com/PyCQA/pydocstyle/issues/635

For example, I fail to make something like this pass our precommit

```
@overload __add__(self,other:float)-> float: ...

@overload __add__(self,other:int)-> int: ...

def __add__(self,other:int|float)->int|float
''' Add an int or float. '''
```

which I would like to use..

**Comment by fzimmermann89** [comment]
Also, I would vote to ignore D105
Docstrings on magic methods in magic methods are not that useful in most cases

**Comment by schuenke** [comment]
Is this solved by #193 ?
